lib/ai/chat_test.go

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai
+
+import (
+	"context"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/sashabaranov/go-openai"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/lib/ai/model"
+	aitest "github.com/gravitational/teleport/lib/ai/testutils"
+)
+
+func TestChat_PromptTokens(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name     string
+		messages []openai.ChatCompletionMessage
+
+		want    int
+		wantErr bool
+	}{
+		{
+			name:     "empty",
+			messages: []openai.ChatCompletionMessage{},
+			want:     0,
+		},
+		{
+			name: "only system message",
+			messages: []openai.ChatCompletionMessage{
+				{
+					Role:    openai.ChatMessageRoleSystem,
+					Content: "Hello",
+				},
+			},
+			want: 44,
```

Comment:
@xacrimon Is it something wrong with this code? It's always returning the same value for used tokens.

Reply:
Yes, the test case loop is only ever looking at completion tokens. These are kept seperate because they are charged seperately and mean different things in the API speak. You probably want to sum the `Prompt` and `Completion` fields on `TokensUsed`

Reply:
Simple fix, applied.